ansible_collections/arista/avd/roles/eos_designs/python_modules/overlay/router_path_selection.py

```diff
@@ -108,35 +103,83 @@ def _get_vrfs(self) -> list | None:
             return vrfs
         return None
 
-    def _get_carrier_id(self, carrier: dict) -> int:
+    def _get_path_group_id(self, path_group_name: str, config_id: int | None = None) -> int:
         """
         TODO - implement algorithm to auto assign IDs - cf internal documenation
         TODO - also implement algorithm for cross connects on public path_groups
         """
-        if carrier["name"] == "LAN_HA":
+        if path_group_name == "LAN_HA":
             return 65535
+        if config_id is not None:
+            return config_id
         return 500
 
-    def _get_local_interfaces(self, carrier: dict) -> list | None:
+    def _get_local_interfaces(self, path_group_name: str) -> list | None:
         """
         Generate the router_path_selection.local_interfaces list
 
         For AUTOVPN clients, configure the stun server profiles as appropriate
         """
         local_interfaces = []
+        for wan_interface in self.shared_utils.wan_interfaces:
+            # Skipping interface not in the target path_group
+            if self.shared_utils.get_carrier_path_group(wan_interface.get("wan_carrier")) != path_group_name:
+                continue
+
+            local_interface = {"name": get(wan_interface, "interface", required=True)}
+
+            if self.shared_utils.wan_role == "client":
+                stun_server_profiles = []
+                for wrr, data in self._wan_route_servers.items():
+                    for path_group in get(data, "wan_path_groups", required=True):
+                        if path_group.get("name") != path_group_name:
+                            continue
+                        if not self._should_connect_to_wan_rr([path_group["name"]]):
+                            continue
+                        for index in range(len(get(path_group, "interfaces", required=True))):
+                            stun_server_profiles.append(self._stun_server_profile_name(wrr, path_group_name, index))
```

Comment:
Dangerous to use index of something to generate names. It will rename things if you remove an earlier entry. Let's discuss how to solve this.

Reply:
in this case it is safe but let's discuss this on a call as it is linked to another ocmment